bridge/launchpad/import.go

```diff
@@ -0,0 +1,80 @@
+package launchpad
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/MichaelMure/git-bug/bridge/core"
+	"github.com/MichaelMure/git-bug/bug"
+	"github.com/MichaelMure/git-bug/cache"
+)
+
+type launchpadImporter struct {
+	conf core.Configuration
+}
+
+func (li *launchpadImporter) Init(conf core.Configuration) error {
+	li.conf = conf
+	return nil
+}
+
+const keyLaunchpadID = "launchpad-id"
+
+func (li *launchpadImporter) makePerson(owner LPPerson) bug.Person {
+	return bug.Person{
+		Name:      owner.Name,
+		Email:     "",
+		Login:     owner.Login,
+		AvatarUrl: "",
+	}
+}
+
+func (li *launchpadImporter) ImportAll(repo *cache.RepoCache) error {
+	lpAPI := new(launchpadAPI)
+
+	err := lpAPI.Init()
+	if err != nil {
+		return err
+	}
+
+	lpBugs, err := lpAPI.SearchTasks(li.conf["project"])
+	if err != nil {
+		return err
+	}
+
+	for _, lpBug := range lpBugs {
+		lpBugID := fmt.Sprintf("%d", lpBug.ID)
+		_, err := repo.ResolveBugCreateMetadata(keyLaunchpadID, lpBugID)
+		if err != nil && err != bug.ErrBugNotExist {
+			return err
+		}
+
+		if err == bug.ErrBugNotExist {
+			createdAt, _ := time.Parse(time.RFC3339, lpBug.CreatedAt)
+			_, err := repo.NewBugRaw(
+				li.makePerson(lpBug.Owner),
+				createdAt.Unix(),
+				lpBug.Title,
+				lpBug.Description,
+				nil,
+				map[string]string{
+					keyLaunchpadID: lpBugID,
+				},
+			)
+			if err != nil {
+				fmt.Println("Failed to add bug #", lpBugID)
```

Comment:
Some nitpicking ;)

- most likely, you want to stop the process and return the error here, because something went very wrong, so better stop before corrupting the local data somehow
- instead of manually piecing together you error message, you can use the `errors.wrap()` function,  which also have the side effect of keeping a stacktrace internally for easier debugging
- error message in go usually start with a lowercase letter, no formatting

so here that would be `return errors.Wrapf(err, "failed to add bug id #%s", lpBugID)`

Reply:
I agree that we should probably just stop if something goes wrong. I used errors.Wrapf in the new version.